### PR TITLE
Support symlinking syntax on NetBSD in ReparsePointUtilities.cs

### PR DESCRIPTION
**Pull request by krytarowski**

### CODE REVIEW

File: src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs [hunk 1]
```diff
@@ -10,6 +10,8 @@ public struct OSPlatform : IEquatable<OSPlatform>
 
         public static OSPlatform Linux { get; } = new OSPlatform("LINUX");
 
+        public static OSPlatform NetBSD { get; } = new OSPlatform("NETBSD");
```
Comment (ellismg) [inline src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs]: The general feeling from the team was we'd like platforms to bake for a while before getting named constants this file (this is consistent with what we are doing with FreeBSD, who's porting effort stalled out) .  For now, can you revert this change and just use  `OSPlatform.Create("NETBSD")` anywhere you need it?

Comment (krytarowski) [inline src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs]: This landed here accidentally. Reverting.

File: src/System.IO.FileSystem/tests/PortedCommon/ReparsePointUtilities.cs [hunk 2]
```diff
@@ -39,7 +39,9 @@ public static bool CreateSymbolicLink(string linkPath, string targetPath)
         {
             symLinkProcess = Process.Start("cmd", string.Format("/c mklink \"{0}\" \"{1}\"", linkPath, targetPath));
         }
-        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ||
+                 RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD")) ||
+                 RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/PortedCommon/ReparsePointUtilities.cs]: Could you just change this to be an else rather than an else if?  If there's ever a non-Windows platform we deal with that doesn't have ln with this syntax, we can deal with it then.

Comment (krytarowski) [inline src/System.IO.FileSystem/tests/PortedCommon/ReparsePointUtilities.cs]: OK!

**Comment by krytarowski** [comment]
Done!

**Comment by stephentoub** [comment]
LGTM. Thanks.
